### PR TITLE
DOCSP-44431 - Add null type

### PR DESCRIPTION
**Pull request by mongoKart**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-44431
Staging - https://deploy-preview-29--docs-pymongo-arrow.netlify.app/data-types/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/data-types.txt [hunk 1]
```diff
@@ -35,29 +35,47 @@ Support for additional types will be added in subsequent releases.
...
-     - ``py.str``, an instance of ``pyarrow.string``
+     - - ``py.str``
+       - Instance of ``pyarrow.string``
    * - Embedded document
-     - ``py.dict``, and instance of ``pyarrow.struct``
+     - - ``py.dict``
+       - Instance of ``pyarrow.struct``
    * - Embedded array
-     - An instance of ``pyarrow.list_``
+     - Instance of ``pyarrow.list_``
```
Comment (jordan-smith721) [inline source/data-types.txt]:
```suggestion
     - - Instance of ``pyarrow.list_``
```

File: source/data-types.txt [hunk 2]
```diff
@@ -35,29 +35,47 @@ Support for additional types will be added in subsequent releases.
...
-     - An instance of ``~pyarrow.bool_``, ``~py.bool``
+     - - ``~py.bool``
+       - Instance of ``~pyarrow.bool_``
    * - 64-bit binary floating point
-     - ``py.float``, an instance of ``pyarrow.float64``
+     - - ``py.float``
+       - Instance of ``pyarrow.float64``
    * - 32-bit integer
-     - An instance of ``pyarrow.int32``
+     - Instance of ``pyarrow.int32``
```
Comment (jordan-smith721) [inline source/data-types.txt]:
```suggestion
     - - Instance of ``pyarrow.int32``
```

File: source/data-types.txt [hunk 3]
```diff
@@ -35,29 +35,47 @@ Support for additional types will be added in subsequent releases.
...
+     - - ``bson.Binary``
+       - Instance of ``pymongoarrow.types.BinaryType``
+       - Instance of ``pymongoarrow.pandas_types.PandasBinary``
    * - JavaScript code
-     - ``bson.Code``, an instance of ``pymongoarrow.types.CodeType``, an instance of ``pymongoarrow.pandas_types.PandasCode``
+     - - ``bson.Code``
+       - Instance of ``pymongoarrow.types.CodeType``
+       - Instance of ``pymongoarrow.pandas_types.PandasCode``
+   * - null
+     - Instance of ``pyarrow.null``
```
Comment (jordan-smith721) [inline source/data-types.txt]:
```suggestion
     - - Instance of ``pyarrow.null``
```

**Review by jordan-smith721 — APPROVED** [review]
LGTM after some formatting fixes

**Review by blink1073 — APPROVED** [review]
LGTM!

**Pull request merged**